packages/browser-wallet/src/background/web3Id.ts

```diff
@@ -198,6 +232,43 @@ export const runIfValidWeb3IdProof: RunCondition<MessageStatusWrapper<undefined>
     }
     try {
         const statements: CredentialStatements = parse(msg.payload.statements);
+
+        // The `verifyAtomicStatements` method only verifies the bounds of the statement variables when it has the
+        // schema available. So we manually do this check here, even though it ideally be moved to the SDK.
+        for (const stat of statements) {
+            for (const atomicStatement of stat.statement) {
+                if (atomicStatement.type === StatementTypes.AttributeInRange) {
+                    const lowerValidationResult = validateAttributeBounds(
+                        atomicStatement.attributeTag,
+                        atomicStatement.lower
+                    );
+                    if (lowerValidationResult.error) {
+                        return rejectRequest(lowerValidationResult.message);
+                    }
+
+                    const upperValidationResult = validateAttributeBounds(
+                        atomicStatement.attributeTag,
+                        atomicStatement.upper
+                    );
+                    if (upperValidationResult.error) {
+                        return rejectRequest(upperValidationResult.message);
+                    }
+                }
+
+                if (
+                    StatementTypes.AttributeInSet === atomicStatement.type ||
+                    StatementTypes.AttributeNotInSet === atomicStatement.type
+                ) {
+                    for (const setItem of atomicStatement.set) {
+                        const validationResult = validateAttributeBounds(atomicStatement.attributeTag, setItem);
+                        if (validationResult.error) {
+                            return rejectRequest(validationResult.message);
+                        }
+                    }
+                }
+            }
+        }
+
         // If a statement does not verify, an error is thrown.
         statements.every((credStatement) =>
```

Comment:
`isValidStringAttribute` in the SDK throws an Error "Buffer" is not defined.

Reply:
Ah, missing web buffer import 😢